src/rfc/ShipItArguments.php

```diff
@@ -0,0 +1,46 @@
+<?hh // strict
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+namespace Facebook\ShipIt;
+
+use namespace HH\Lib\{Vec, Str};
+
+interface IShipItArgumentParser {
+  public function parseArgs(
+    vec<ShipItCLIArgument> $config,
+  ): dict<string, mixed>;
+}
+
+class ShipItInvocation implements IShipItArgumentParser {
+  use ShipItArgumentsTrait;
+
+  public function parseArgs(
+    vec<ShipItCLIArgument> $config,
+  ): dict<string, mixed> {
+    return $this->getArgs();
+  }
+
+  public function run(): void {
+    ShipDemoProject::cliMain($this);
+  }
+}
+
+class ShipItCLIArgumentParser implements IShipItArgumentParser {
+  public function parseArgs(
+    vec<ShipItCLIArgument> $config,
+  ): dict<string, mixed> {
+    /* HH_IGNORE_ERROR[2049] __PHPStdLib */
+    /* HH_IGNORE_ERROR[4107] __PHPStdLib */
+    return \getopt(
```

Comment:
Is it possible to use hh-clilib instead of getopt? That will solve existing problems with poor handling of unrecognized args.

Reply:
Sure I'll look into it

Reply:
@yns88 I believe it can be done, but I might take a stab at it in another PR.

I think the best idea would be to delete `FBShipItCLIArgument` entirely and replace each instance with hh-cli's [`CLIOptions\*`](https://github.com/hhvm/hh-clilib#options) at the phase level, which sounds out-of-scope to me for this PR. Happy to do it here if you'd prefer.